OpenRA.Platforms.Default/Sdl2GraphicsContext.cs

```diff
@@ -111,39 +115,51 @@ public void DisableScissor()
 			OpenGL.CheckGLError();
 		}
 
-		public Bitmap TakeScreenshot()
+		public void SaveScreenshot(string path)
 		{
-			var rect = new Rectangle(Point.Empty, window.SurfaceSize);
-			var bitmap = new Bitmap(rect.Width, rect.Height, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
-			var data = bitmap.LockBits(rect,
-				System.Drawing.Imaging.ImageLockMode.WriteOnly, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+			var s = window.SurfaceSize;
+			var raw = new byte[s.Width * s.Height * 4];
 
-			OpenGL.glPushClientAttrib(OpenGL.GL_CLIENT_PIXEL_STORE_BIT);
-
-			OpenGL.glPixelStoref(OpenGL.GL_PACK_ROW_LENGTH, data.Stride / 4f);
-			OpenGL.glPixelStoref(OpenGL.GL_PACK_ALIGNMENT, 1);
+			unsafe
+			{
+				fixed (byte* pRaw = raw)
+				{
+					OpenGL.glPushClientAttrib(OpenGL.GL_CLIENT_PIXEL_STORE_BIT);
```

Comment:
I would prefer to keep the unsafe/pinned block as small as possible. This state mangling doesn't need to be inside.